pulsar/consumer_partition.go

```diff
@@ -1190,19 +1193,26 @@ func getPreviousMessage(mid trackingMessageID) trackingMessageID {
 }
 
 func (pc *partitionConsumer) Decompress(msgMeta *pb.MessageMetadata, payload internal.Buffer) (internal.Buffer, error) {
-	pc.providersMutex.RLock()
-	provider, ok := pc.compressionProviders[msgMeta.GetCompression()]
-	pc.providersMutex.RUnlock()
+	v, ok := pc.compressionProviders.Load(msgMeta.GetCompression())
```

Comment:
Maybe we can replace `v` with meaningful variable names

Reply:
Sure thing.

Reply:
@wolfstudy , I've renamed the variables in this method for additional clarity.

Reply:
Cool, nice work